pyclean/py3clean.py

```diff
@@ -2,175 +2,100 @@
-        def find_files_to_remove(pyfile):  # NOQA
-            directory = join(dirname(pyfile), '__pycache__')
-            fname = splitext(basename(pyfile))[0]
-            for fn in glob1(directory, "%s.%s.py[co]" % (fname, magic_tag)):
-                yield join(directory, fn)
-            # workaround for http://bugs.python.org/issue22966
-            if '.' in fname:
-                sane_fname = join(dirname(pyfile), fname.split('.', 1)[0])
-                for fn in find_files_to_remove(sane_fname):
-                    yield join(directory, fn)
-
-    def myremove(fname):
-        remove(fname)
-        directory = dirname(fname)
-        # remove __pycache__ directory if it's empty
-        if directory.endswith('__pycache__'):
-            try:
-                rmdir(directory)
-            except Exception:
-                pass
-
-    counter = 0
-    try:
-        while True:
-            pyfile = (yield)
-            for filename in find_files_to_remove(pyfile):
-                try:
-                    myremove(filename)
-                    counter += 1
-                except (IOError, OSError) as e:
-                    log.error('cannot remove %s', filename)
-                    log.debug(e)
-    except GeneratorExit:
-        log.info("removed files: %s", counter)
+    parser = argparse.ArgumentParser(description="Removes .pyc files and __pycache__ directories from a tree")
+    parser.add_argument("root_dir",
+                        metavar="ROOT_PATH",
+                        help="Path to start searching for file in.")
+    return parser    
 
-
-def main(args):
+def main(args: argparse.Namespace) -> None:
     """Entry point for Python 3"""
-    if args.verbose or environ.get('PYCLEAN_DEBUG') == '1':
-        log.setLevel(logging.DEBUG)
-        log.debug('argv: %s', sys.argv)
-        log.debug('args: %s', args)
-    else:
-        log.setLevel(logging.WARNING)
-
-    if args.version:
-        if args.version.endswith('3.1'):  # 3.1, -3.1
-            magic_tag = False
-        else:
-            magic_tag = get_magic_tag_to_remove(getver(args.version))
-        d = destroyer(magic_tag)
-    else:
-        d = destroyer()  # remove everything
-    next(d)  # initialize coroutine
-
-    if args.package:
-        for pkg in args.package:
-            log.info('cleaning package %s', pkg)
-            pfiles = set(dpf.from_package(pkg))
-
-    if args.directory:
-        log.info('cleaning directories: %s', args.directory)
-        files = set(dpf.from_directory(args.directory))
-        if args.package:
-            files = files & pfiles
-    else:
-        files = pfiles
-
-    for filename in files:
-        d.send(filename)
+    dirs_to_check, files_to_remove, dirs_to_remove = find_files(args.root_dir)
+    print(f"Found {len(files_to_remove)} files to remove in {len(dirs_to_check)} directories. Can remove {len(dirs_to_remove)} directories.")
+    remove_files_and_dirs(files_to_remove, dirs_to_remove)
```

Comment:
It seems like adding these three lines (plus the two methods therein, obviously) to the [modern](https://github.com/bittner/pyclean/blob/master/pyclean/modern.py) module would be sufficient.

Do you think you have time to do this in the next couple of days?